kernel/kernel-api/src/main/java/io/delta/kernel/internal/data/TransactionStateRow.java

```diff
@@ -68,6 +73,30 @@ public static StructType getLogicalSchema(Engine engine, Row transactionState) {
         serializedSchema);
   }
 
+  /**
+   * Get the configuration from the transaction state {@link Row} returned by {@link
+   * Transaction#getTransactionState(Engine)}
+   *
+   * @param transactionState
+   * @return Configuration as a map of key-value pairs.
+   */
+  public static Map<String, String> getConfiguration(Row transactionState) {
+    return VectorUtils.toJavaMap(transactionState.getMap(COL_NAME_TO_ORDINAL.get("configuration")));
+  }
+
+  /**
+   * Get the iceberg compatibility enabled or not from the transaction state {@link Row} returned by
+   * {@link Transaction#getTransactionState(Engine)}
+   *
+   * @param transactionState Transaction state state {@link Row}
+   * @return True if iceberg compatibility is enabled, false otherwise.
+   */
+  public static boolean isIcebergCompatV2Enabled(Row transactionState) {
+    return Boolean.parseBoolean(
```

Comment:
Sorry why doesn't this use the updated `fromMetadata` in TableConfig?

Reply:
Currently it requires an `Engine`. It was added because of the Coordinated commits requiring a JSON parser to parse a table config. I have clean up task to remove the `Engine` requirement for `fromMetadata`.